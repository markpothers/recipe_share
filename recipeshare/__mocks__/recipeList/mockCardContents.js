export const listProps = [
	{
		"testTitle": "default",
		"chef_commented": 2,
		"chef_liked": 1,
		"chef_made": 0,
		"chef_shared": 0,
		"chefimage_url": "",
		"comments_count": 2,
		"difficulty": 4,
		"id": 306,
		"image_url": null,
		"likes_count": 1,
		"makes_count": 0,
		"name": "Test recipe",
		"renderOfflineMessage": [],
		"shared_id": 306,
		"sharer_id": 97,
		"sharer_username": "pothers",
		"shares_count": 1,
		"time": "01:00",
		"username": "pothers",
	},
	{
		"testTitle": "longStrings",
		"chef_commented": 2,
		"chef_liked": 1,
		"chef_made": 0,
		"chef_shared": 0,
		"chefimage_url": "",
		"comments_count": 2,
		"difficulty": 4,
		"id": 307,
		"image_url": null,
		"likes_count": 1,
		"makes_count": 0,
		"name": "This is my test recipe with a really long title and I want to know that it renders properly when the title string is really long.",
		"renderOfflineMessage": [],
		"shared_id": 306,
		"sharer_id": 97,
		"sharer_username": "pothers",
		"shares_count": 1,
		"time": "01:00",
		"username": "pothers",
	},
]

