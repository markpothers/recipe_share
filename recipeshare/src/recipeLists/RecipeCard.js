import React from 'react'
import { Image, View, TouchableOpacity, Text } from 'react-native'
import { styles } from './recipeListStyleSheet'
import { databaseURL } from '../dataComponents/databaseURL'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { responsiveWidth, responsiveHeight, responsiveFontSize } from 'react-native-responsive-dimensions'; //eslint-disable-line no-unused-vars
import OfflineMessage from '../offlineMessage/offlineMessage'
export default class RecipeCard extends React.PureComponent {

	navigateToSharer = (chefID) => {
		this.props.navigateToChefDetails(chefID, this.props.id)
	}

	render() {
		// console.log(this.props)
		return (
			<View style={styles.recipeCard} >
				{this.props.renderOfflineMessage.includes(this.props.id) && (
					<OfflineMessage
						message={`Sorry, can't do that right now.${"\n"}You appear to be offline.`}
						topOffset={'35%'}
						clearOfflineMessage={() => { this.props.clearOfflineMessage(this.props.id) }}
					/>)}
				{this.props.sharer_id && <PostedBy navigateToSharer={this.navigateToSharer} username={this.props.sharer_username} sharer_id={this.props.sharer_id} />}
				<View style={styles.recipeCardTopContainer}>
					<View style={styles.recipeCardTopLeftContainer}>
						<TouchableOpacity style={styles.recipeCardTopLeftUpperContainer} activeOpacity={0.7} onPress={() => this.props.navigateToRecipeDetails(this.props.id)}>
							<Text maxFontSizeMultiplier={2} style={styles.recipeCardHighlighted}>{this.props.name}</Text>
						</TouchableOpacity>
						<View style={styles.recipeCardTopLeftMiddleContainer}>
							<Text maxFontSizeMultiplier={2} style={styles.recipeCardTopItalic}>Created by: </Text>
							<TouchableOpacity onPress={() => this.props.navigateToChefDetails(this.props.chef_id, this.props.id)}>
								<Text maxFontSizeMultiplier={2} style={styles.recipeCardTopItalic}>{this.props.username}</Text>
							</TouchableOpacity>
						</View>
						<View style={styles.recipeCardTopLeftLowerContainer}>
							<Text maxFontSizeMultiplier={2} style={styles.recipeCardTopOther}>Prep time: {this.props.time}</Text>
							<Text maxFontSizeMultiplier={2} style={styles.recipeCardTopOther}>Difficulty: {this.props.difficulty}/10</Text>
						</View>
					</View>
					<TouchableOpacity style={styles.recipeCardTopRightContainer} onPress={() => this.props.navigateToChefDetails(this.props.chef_id, this.props.id)}>
						<AvatarImage chefimage_url={this.props.chefimage_url} />
					</TouchableOpacity>
				</View>
				<TouchableOpacity style={styles.recipeCardImageContainer} activeOpacity={0.7} onPress={() => this.props.navigateToRecipeDetails(this.props.id)}>
					<Image style={styles.thumbnail} source={{ uri: this.props.image_url }} />
				</TouchableOpacity>
				<View style={styles.recipeCardBottomContainer}>
					<TouchableOpacity style={styles.recipeCardBottomSubContainers} onPress={this.props.chef_shared == 0 ? () => this.props.reShareRecipe(this.props.id) : () => this.props.unReShareRecipe(this.props.id)}>
						{this.props.chef_shared == 0 ? <Icon name='share-outline' size={responsiveHeight(3.5)} style={styles.icon} /> : <Icon name='share' size={responsiveHeight(3.5)} style={styles.icon} />}
						<Text maxFontSizeMultiplier={2} style={styles.recipeCardBottomOther} >{this.props.shares_count}</Text>
					</TouchableOpacity>
					<TouchableOpacity style={styles.recipeCardBottomSubContainers} onPress={this.props.chef_liked == 0 ? () => this.props.likeRecipe(this.props.id) : () => this.props.unlikeRecipe(this.props.id)}>
						{this.props.chef_liked == 0 ? <Icon name='heart-outline' size={responsiveHeight(3.5)} style={styles.icon} /> : <Icon name='heart' size={responsiveHeight(3.5)} style={styles.icon} />}
						<Text maxFontSizeMultiplier={2} style={styles.recipeCardBottomOther} >{this.props.likes_count}</Text>
					</TouchableOpacity>
					{/* <TouchableOpacity style={styles.recipeCardBottomSubContainers} onPress={this.props.chef_made === 0 ? () => this.props.makeRecipe(this.props.id) : null }>
                            {this.props.chef_made === 0 ? <Icon name='food' size={responsiveHeight(3.5)} style={styles.icon}/> : <Icon name='food-off' size={responsiveHeight(3.5)} style={styles.icon}/> }
                            <Text style={styles.recipeCardBottomOther} >{this.props.makes_count}</Text>
                        </TouchableOpacity> */}
					<TouchableOpacity style={styles.recipeCardBottomSubContainers} onPress={() => this.props.navigateToRecipeDetails(this.props.id, true)}>
						{this.props.chef_commented == 0 ? <Icon name='comment-outline' size={responsiveHeight(3.5)} style={styles.icon} /> : <Icon name='comment' size={responsiveHeight(3.5)} style={styles.icon} />}
						<Text maxFontSizeMultiplier={2} style={styles.recipeCardBottomOther} >{this.props.comments_count}</Text>
					</TouchableOpacity>
				</View>
			</View>
		)
	}
}

function AvatarImage(chefimage_url) {
	const URL = chefimage_url.chefimage_url
	if (URL == null) {
		return (
			<Image style={styles.avatarThumbnail} source={require("../dataComponents/peas.jpg")} />
		)
	} else {
		return (
			<Image style={styles.avatarThumbnail} source={URL.startsWith("http") ? { uri: URL } : { uri: `${databaseURL}${URL}` }} />
		)
	}
}

function PostedBy(props) {
	return (
		<View style={styles.recipeCardTopPostedByContainer}>
			<Text maxFontSizeMultiplier={2} style={styles.recipeCardTopItalic}>Re-shared by: </Text>
			<TouchableOpacity onPress={() => props.navigateToSharer(props.sharer_id)}>
				<Text maxFontSizeMultiplier={2} style={styles.recipeCardTopItalic} >{props.username}</Text>
			</TouchableOpacity>
		</View>
	)
}
