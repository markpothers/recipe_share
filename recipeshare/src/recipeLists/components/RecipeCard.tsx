import { Image, Text, TouchableOpacity, View } from "react-native";
import { responsiveHeight, responsiveWidth } from "react-native-responsive-dimensions";

import Icon from "react-native-vector-icons/MaterialCommunityIcons";
import OfflineMessage from "../../offlineMessage/offlineMessage";
import React from "react";
import { getTimeStringFromMinutes } from "../../auxFunctions/getTimeStringFromMinutes";
import { styles } from "../recipeListStyleSheet";

// eslint-disable-next-line @typescript-eslint/no-var-requires
const defaultRecipeImage = require("../../dataComponents/default-recipe.jpg");

type OwnProps = {
	navigateToChefDetails: (chefID: number, recipeID: number) => void;
	navigateToRecipeDetails: (recipeID: number, commenting?: boolean) => void;
	id: number;
	image_url?: string;
	renderOfflineMessage?: number[];
	clearOfflineMessage: (recipeID: number) => void;
	sharer_id?: number;
	sharer_username: string;
	chef_id: number;
	username: string;
	total_time: number;
	difficulty: number;
	chefimage_url: string;
	chef_shared: number;
	reShareRecipe: (recipeID: number) => void;
	unReShareRecipe: (recipeID: number) => void;
	shares_count: number;
	chef_liked: number;
	likeRecipe: (recipeID: number) => void;
	unlikeRecipe: (recipeID: number) => void;
	likes_count: number;
	chef_made: number;
	makeRecipe: (recipeID: number) => void;
	makes_count: number;
	chef_commented: number;
	comments_count: number;
	name: string;
};

const RecipeCard = ({
	navigateToChefDetails,
	navigateToRecipeDetails,
	id,
	image_url,
	renderOfflineMessage,
	clearOfflineMessage,
	sharer_username,
	sharer_id,
	chef_id,
	username,
	total_time,
	difficulty,
	chefimage_url,
	chef_shared,
	reShareRecipe,
	unReShareRecipe,
	shares_count,
	chef_liked,
	likeRecipe,
	unlikeRecipe,
	likes_count,
	// chef_made,
	// makeRecipe,
	// makes_count,
	chef_commented,
	comments_count,
	name,
}: OwnProps) => {
	// export default class RecipeCard extends React.PureComponent {
	const navigateToSharer = (chefID) => {
		navigateToChefDetails(chefID, id);
	};

	// render() {
	const imageSource = image_url ? { uri: image_url } : defaultRecipeImage;
	return (
		<View style={styles.recipeCard} testID={"recipeCard"}>
			{renderOfflineMessage?.includes(id) && (
				<OfflineMessage
					message={`Sorry, can't do that right now.${"\n"}You appear to be offline.`}
					topOffset={"35%"}
					clearOfflineMessage={() => {
						clearOfflineMessage(id);
					}}
				/>
			)}
			{sharer_id && (
				<PostedBy navigateToSharer={navigateToSharer} username={sharer_username} sharer_id={sharer_id} />
			)}
			<View style={styles.recipeCardTopContainer}>
				<View style={styles.recipeCardTopLeftContainer}>
					<TouchableOpacity
						testID={"recipeNameButton"}
						style={styles.recipeCardTopLeftUpperContainer}
						activeOpacity={0.7}
						onPress={() => navigateToRecipeDetails(id)}
					>
						<Text maxFontSizeMultiplier={2} style={styles.recipeCardHighlighted}>
							{name}
						</Text>
					</TouchableOpacity>
					<View style={styles.recipeCardTopLeftMiddleContainer}>
						<Text maxFontSizeMultiplier={2} style={styles.recipeCardTopItalic}>
							Created by:{" "}
						</Text>
						<TouchableOpacity testID={"chefNameButton"} onPress={() => navigateToChefDetails(chef_id, id)}>
							<Text maxFontSizeMultiplier={2} style={styles.recipeCardTopItalic}>
								{username}
							</Text>
						</TouchableOpacity>
					</View>
					<View style={styles.recipeCardTopLeftLowerContainer}>
						<Text maxFontSizeMultiplier={2} style={styles.recipeCardTopOther}>
							Total time: {getTimeStringFromMinutes(total_time)}
						</Text>
						<Text maxFontSizeMultiplier={2} style={styles.recipeCardTopOther}>
							Difficulty: {difficulty}/10
						</Text>
					</View>
				</View>
				<TouchableOpacity
					testID={"chefImageButton"}
					style={styles.recipeCardTopRightContainer}
					onPress={() => navigateToChefDetails(chef_id, id)}
				>
					<AvatarImage chefimage_url={chefimage_url} />
				</TouchableOpacity>
			</View>
			<TouchableOpacity
				testID={"recipeImageButton"}
				style={styles.recipeCardImageContainer}
				activeOpacity={0.7}
				onPress={() => navigateToRecipeDetails(id)}
				accessibilityLabel={"picture of recipe"}
			>
				<Image style={styles.thumbnail} source={imageSource} />
			</TouchableOpacity>
			<View style={styles.recipeCardBottomContainer}>
				<TouchableOpacity
					testID={"reShareButton"}
					style={styles.recipeCardBottomSubContainers}
					onPress={chef_shared == 0 ? () => reShareRecipe(id) : () => unReShareRecipe(id)}
				>
					{chef_shared == 0 ? (
						<Icon
							name="share-outline"
							size={responsiveHeight(3.5)}
							style={styles.icon}
							accessibilityLabel={"share recipe with followers"}
						/>
					) : (
						<Icon
							name="share"
							size={responsiveHeight(3.5)}
							style={styles.icon}
							accessibilityLabel={"remove share"}
						/>
					)}
					<Text
						maxFontSizeMultiplier={2}
						style={styles.recipeCardBottomOther}
						accessibilityLabel={"shares count"}
					>
						{shares_count}
					</Text>
				</TouchableOpacity>
				<TouchableOpacity
					testID={"likeButton"}
					style={styles.recipeCardBottomSubContainers}
					onPress={chef_liked == 0 ? () => likeRecipe(id) : () => unlikeRecipe(id)}
				>
					{chef_liked == 0 ? (
						<Icon
							name="heart-outline"
							size={responsiveHeight(3.5)}
							style={styles.icon}
							accessibilityLabel={"like recipe"}
						/>
					) : (
						<Icon
							name="heart"
							size={responsiveHeight(3.5)}
							style={styles.icon}
							accessibilityLabel={"unlike recipe"}
						/>
					)}
					<Text
						maxFontSizeMultiplier={2}
						style={styles.recipeCardBottomOther}
						accessibilityLabel={"likes count"}
					>
						{likes_count}
					</Text>
				</TouchableOpacity>
				{/* <TouchableOpacity style={styles.recipeCardBottomSubContainers} onPress={chef_made === 0 ? () => makeRecipe(id) : null }>
                            {chef_made === 0 ? <Icon name='food' size={responsiveHeight(3.5)} style={styles.icon}/> : <Icon name='food-off' size={responsiveHeight(3.5)} style={styles.icon}/> }
                            <Text style={styles.recipeCardBottomOther} >{makes_count}</Text>
                        </TouchableOpacity> */}
				<TouchableOpacity
					testID={"commentButton"}
					style={styles.recipeCardBottomSubContainers}
					onPress={() => navigateToRecipeDetails(id, true)}
				>
					{chef_commented == 0 ? (
						<Icon
							name="comment-outline"
							size={responsiveHeight(3.5)}
							style={styles.icon}
							accessibilityLabel={"comment on recipe"}
						/>
					) : (
						<Icon
							name="comment"
							size={responsiveHeight(3.5)}
							style={styles.icon}
							accessibilityLabel={"comment on recipe"}
						/>
					)}
					<Text
						maxFontSizeMultiplier={2}
						style={styles.recipeCardBottomOther}
						accessibilityLabel={"comments count"}
					>
						{comments_count}
					</Text>
				</TouchableOpacity>
			</View>
		</View>
	);
};

export default RecipeCard;

function AvatarImage(chefimage_url: { chefimage_url: string }) {
	const URL = chefimage_url.chefimage_url;
	if (!URL) {
		return (
			<Image
				style={styles.avatarThumbnail}
				source={require("../../dataComponents/default-chef.jpg")}
				accessibilityLabel={"picture of chef"}
			/>
		);
	} else {
		return <Image style={styles.avatarThumbnail} source={{ uri: URL }} accessibilityLabel={"picture of chef"} />;
	}
}

type PostedByProps = {
	navigateToSharer: (chefID: number) => void;
	username: string;
	sharer_id: number;
};

function PostedBy({ navigateToSharer, sharer_id, username }: PostedByProps) {
	return (
		<View testID={"postedByElement"} style={styles.recipeCardTopPostedByContainer}>
			<Icon name="share" size={responsiveHeight(3.5)} style={styles.reSharedIcon} />
			<Text maxFontSizeMultiplier={2} style={[styles.recipeCardTopItalic, { maxWidth: responsiveWidth(28) }]}>
				Re-shared by:{" "}
			</Text>
			<TouchableOpacity
				testID={"sharerNameButton"}
				style={styles.recipeCardTopPostedByTouchable}
				onPress={() => navigateToSharer(sharer_id)}
			>
				<Text maxFontSizeMultiplier={2} style={styles.recipeCardTopItalic}>
					{username}
				</Text>
			</TouchableOpacity>
		</View>
	);
}
