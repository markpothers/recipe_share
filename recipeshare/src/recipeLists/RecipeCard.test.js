import 'react-native'
import React from 'react';
import renderer from 'react-test-renderer';

import RecipeCard from './RecipeCard.js'
import { listProps } from '../../__mocks__/recipeList/mockCardContents.js'

describe('Recipe Card', () => {

	let mountedRecipeCard

	beforeAll(() => {
		// console.log('runs at the beginning of everything')
	})

	beforeEach(() => {
		// console.log('runs before every test')
	})

	afterEach(() => {
		// console.log('runs after each test')
	})

	afterAll(() => {
		// console.log('runs after all tests have completed')
	})

	// it('asyncly works', done => {
	// 	setTimeout(() => {
	// 		//do things
	// 		done()
	// 	}, 100)
	// })

	it('can be rendered using typical props and matches its previous image', () => {
		const testProps = listProps.find(p => p.testTitle == "default")
		mountedRecipeCard = renderer.create(<RecipeCard {...testProps} />)
		const image = mountedRecipeCard.toJSON()
		expect(image).toMatchSnapshot()
	})

	it('can be rendered using different props including a really long recipe title matches its previous image', () => {
		const testProps = listProps.find(p => p.testTitle == "longStrings")
		mountedRecipeCard = renderer.create(<RecipeCard {...testProps} />)
		const image = mountedRecipeCard.toJSON()
		expect(image).toMatchSnapshot()
	})

});
