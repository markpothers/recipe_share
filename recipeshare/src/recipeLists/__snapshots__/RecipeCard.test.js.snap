// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Card can be rendered using different props including a really long recipe title matches its previous image 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#104e01",
      "borderRadius": 5,
      "borderStyle": "solid",
      "borderWidth": 1,
      "marginBottom": 2.668,
      "marginTop": 2.668,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#104e01",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "marginLeft": "1.6%",
        "marginRight": "2%",
        "paddingBottom": "1%",
        "paddingTop": "1%",
        "width": "96.4%",
      }
    }
  >
    <Text
      maxFontSizeMultiplier={2}
      style={
        Object {
          "color": "#505050",
          "fontSize": 29.06613627191921,
          "fontStyle": "italic",
        }
      }
    >
      Re-shared by: 
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        maxFontSizeMultiplier={2}
        style={
          Object {
            "color": "#505050",
            "fontSize": 29.06613627191921,
            "fontStyle": "italic",
          }
        }
      >
        pothers
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": "1%",
        "marginTop": "1%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 8,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginLeft": "2%",
            "marginRight": "2%",
            "opacity": 1,
            "width": "96%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Object {
              "color": "#505050",
              "fontSize": 32.12572956370018,
              "fontWeight": "bold",
              "textAlign": "left",
            }
          }
        >
          This is my test recipe with a really long title and I want to know that it renders properly when the title string is really long.
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "marginLeft": "2%",
            "marginRight": "2%",
            "width": "96%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Object {
              "color": "#505050",
              "fontSize": 29.06613627191921,
              "fontStyle": "italic",
            }
          }
        >
          Created by: 
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            maxFontSizeMultiplier={2}
            style={
              Object {
                "color": "#505050",
                "fontSize": 29.06613627191921,
                "fontStyle": "italic",
              }
            }
          >
            pothers
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": "2%",
            "marginRight": "2%",
            "width": "96%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Object {
              "color": "#505050",
              "flex": 1,
              "fontSize": 29.06613627191921,
            }
          }
        >
          Prep time: 
          01:00
        </Text>
        <Text
          maxFontSizeMultiplier={2}
          style={
            Object {
              "color": "#505050",
              "flex": 1,
              "fontSize": 29.06613627191921,
            }
          }
        >
          Difficulty: 
          4
          /10
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
          "opacity": 1,
          "overflow": "hidden",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 5,
            "height": 120.06,
            "marginRight": "4%",
            "width": 120.06,
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 500.25,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 5,
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": "1%",
        "marginTop": "1%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text />
      <Text
        maxFontSizeMultiplier={2}
        style={
          Object {
            "color": "#505050",
            "fontSize": 32.12572956370018,
            "fontWeight": "bold",
            "marginLeft": "5%",
            "textAlign": "center",
          }
        }
      >
        1
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text />
      <Text
        maxFontSizeMultiplier={2}
        style={
          Object {
            "color": "#505050",
            "fontSize": 32.12572956370018,
            "fontWeight": "bold",
            "marginLeft": "5%",
            "textAlign": "center",
          }
        }
      >
        1
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text />
      <Text
        maxFontSizeMultiplier={2}
        style={
          Object {
            "color": "#505050",
            "fontSize": 32.12572956370018,
            "fontWeight": "bold",
            "marginLeft": "5%",
            "textAlign": "center",
          }
        }
      >
        2
      </Text>
    </View>
  </View>
</View>
`;

exports[`Recipe Card can be rendered using typical props and matches its previous image 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#104e01",
      "borderRadius": 5,
      "borderStyle": "solid",
      "borderWidth": 1,
      "marginBottom": 2.668,
      "marginTop": 2.668,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#104e01",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "marginLeft": "1.6%",
        "marginRight": "2%",
        "paddingBottom": "1%",
        "paddingTop": "1%",
        "width": "96.4%",
      }
    }
  >
    <Text
      maxFontSizeMultiplier={2}
      style={
        Object {
          "color": "#505050",
          "fontSize": 29.06613627191921,
          "fontStyle": "italic",
        }
      }
    >
      Re-shared by: 
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        maxFontSizeMultiplier={2}
        style={
          Object {
            "color": "#505050",
            "fontSize": 29.06613627191921,
            "fontStyle": "italic",
          }
        }
      >
        pothers
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": "1%",
        "marginTop": "1%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 8,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginLeft": "2%",
            "marginRight": "2%",
            "opacity": 1,
            "width": "96%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Object {
              "color": "#505050",
              "fontSize": 32.12572956370018,
              "fontWeight": "bold",
              "textAlign": "left",
            }
          }
        >
          Test recipe
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "marginLeft": "2%",
            "marginRight": "2%",
            "width": "96%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Object {
              "color": "#505050",
              "fontSize": 29.06613627191921,
              "fontStyle": "italic",
            }
          }
        >
          Created by: 
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            maxFontSizeMultiplier={2}
            style={
              Object {
                "color": "#505050",
                "fontSize": 29.06613627191921,
                "fontStyle": "italic",
              }
            }
          >
            pothers
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": "2%",
            "marginRight": "2%",
            "width": "96%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Object {
              "color": "#505050",
              "flex": 1,
              "fontSize": 29.06613627191921,
            }
          }
        >
          Prep time: 
          01:00
        </Text>
        <Text
          maxFontSizeMultiplier={2}
          style={
            Object {
              "color": "#505050",
              "flex": 1,
              "fontSize": 29.06613627191921,
            }
          }
        >
          Difficulty: 
          4
          /10
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
          "opacity": 1,
          "overflow": "hidden",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 5,
            "height": 120.06,
            "marginRight": "4%",
            "width": 120.06,
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 500.25,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 5,
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": "1%",
        "marginTop": "1%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text />
      <Text
        maxFontSizeMultiplier={2}
        style={
          Object {
            "color": "#505050",
            "fontSize": 32.12572956370018,
            "fontWeight": "bold",
            "marginLeft": "5%",
            "textAlign": "center",
          }
        }
      >
        1
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text />
      <Text
        maxFontSizeMultiplier={2}
        style={
          Object {
            "color": "#505050",
            "fontSize": 32.12572956370018,
            "fontWeight": "bold",
            "marginLeft": "5%",
            "textAlign": "center",
          }
        }
      >
        1
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text />
      <Text
        maxFontSizeMultiplier={2}
        style={
          Object {
            "color": "#505050",
            "fontSize": 32.12572956370018,
            "fontWeight": "bold",
            "marginLeft": "5%",
            "textAlign": "center",
          }
        }
      >
        2
      </Text>
    </View>
  </View>
</View>
`;
